### ISSUE TITLE
Prevent controlled component from jagging

### ISSUE THREAD

**Issue by degroote22** [mariusandra/pigeon-maps#16]
Right now if you have a controlled map and if updates without it being the map fault while animating (perhaps a marker has updated) then the map will go through componentWillReceiveProps and will force the map to go to the last given center and zoom, ignoring the user intention and jagging.
I have come up with a fix but it's quite ugly. I'd like help to refactor and then send a PR. (help pls)
That's the code to be added [here](https://github.com/mariusandra/pigeon-maps/blob/master/src/index.js#L130):

```javascript
      if (!nextProps.center && !nextProps.zoom) {
        // if the user isn't controlling neither zoom nor center we don't have to update.
        return;
      }
      //controllling zoom but zoom didn't change
      if (
        nextProps.zoom === this.props.zoom && !nextProps.center
      ) {
        return;
      }
      //controlling center but center didn't change
      if ( 
        nextProps.center &&
        nextProps.center[0] === this.props.center[0] &&
        nextProps.center[1] === this.props.center[1] && !nextProps.zoom
      ) {
        return;
      }
      //controlling both and neither changed
      if (
        nextProps.center &&
        nextProps.center[0] === this.props.center[0] &&
        nextProps.center[1] === this.props.center[1] &&
        nextProps.zoom === this.props.zoom
      ) {
        return;
      }
```

Also, @mariusandra, can this change brake another part of pigeon-maps? I have been testing this 'fix' for about 10 minutes and it works perfectly.
If someone needs examples of after and before(and what's the issue that happened that I'm talking about) this I can provide.

Thanks for your time.

**Comment by mariusandra** [comment]
Hey, I don't think this will break anything else. 

For comparing then, if the props are not controlled, then shouldn't `nextProps.zoom === this.props.zoom` anyway, as both will be `undefined`?

So just replace all of the above with something like this:

```js
if (
(!nextProps.center || (nextProps.center[0] === this.props.center[0] && nextProps.center[1] === this.props.center[1])) && 
nextProps.zoom === this.props.zoom) {
  return
}
```

**Comment by mariusandra** [comment]
I released `0.8.0`, but didn't include this change. Feel free to send a PR so you can get the credit, and then I'll release `0.8.1` :)
